python/cuda_cccl/tests/parallel/test_scan.py

```diff
@@ -66,12 +72,14 @@ def op(a, b):
     h_init = np.array([42], dtype=dtype)
     d_output = cp.empty_like(d_input)
 
-    scan_device(d_input, d_output, len(d_input), op, h_init, force_inclusive)
+    scan_device(d_input, d_output, len(d_input), reduce_op, h_init, force_inclusive)
 
     got = d_output.get()
     expected = scan_host(d_input.get(), op, h_init, force_inclusive)
 
-    np.testing.assert_allclose(expected, got, rtol=1e-5)
+    np.testing.assert_allclose(
+        expected, got, rtol=0.08 if dtype == np.float16 else 1e-5
+    )
```

Comment:
```suggestion
    if np.isdtype(dtype, ('real floating', 'complex floating'):
        real_dt = np.finfo(dtype).dtype
        eps = np.finfo(real_dt).eps
        rtol =  82 * eps
        np.testing.assert_allclose(expected, got, rtol=rtol)
    else:
        np.testing.assert_array_equal(expected, got)
```

I think this testing makes it clear how many binary ULP (units-in-the-last-place) we tolerate to disagree irrespective of the width of mantissa.